src/os/ui/layer/compare/layercompareui.js

```diff
@@ -493,6 +503,27 @@ const launchLayerCompare = (options) => {
   }
 };
 
+
+/**
+ * Count the features in an array of layers.
+ * @param {Array<Layer>=} layerArray An array of layers.
+ * @return {number} The total number of features in the layer array.
+ */
+const countFeatures = (layerArray) => {
+  if (layerArray && layerArray.length > 0) {
+    var featureCount = 0;
+    layerArray.forEach((layer) => {
+      var source = /** @type {ISource} */ (layer.getSource());
+      if (source) {
+        featureCount += source.getFeatures().length;
```

Comment:
You should first verify the source is an `ISource` or this line will break, for example when comparing tile layers.